src/controllers/events.js

```diff
@@ -23,11 +23,13 @@ exports.getEvent = async (req, res, next) => {
 };
 
 exports.getRelatedEvents = async (req, res, next) => {
-  const event = await Event.findById(req.params.id, { topic: 1 });
+  const event = await Event.findById(req.params.id, { tags: 1 });
   if (!event) {
     return next(new ErrorResponse('No event found', 404));
   }
-  const relatedEvents = await Event.find()
+  const relatedEvents = await Event.find({
+    tags: { $in: [event.tags[0], event.tags[event.tags.length - 1]] },
```

Comment:
This query would be better  `tags: { $in: event.tags },`

Reply:
the query and the related test unit has been enhanced.